### ISSUE TITLE
Update the "Classic ribbon UI" mod to work with latest Windows 11 update

### ISSUE THREAD

**Issue by LesFerch** [ramensoftware/windhawk-mods#485]
After KB5034204 (OS Builds 22621.3085 and 22631.3085) is installed, the "Classic navigation bar" option no longer works.

**Comment by m417z** [comment]
Please attach debug logs by following the instructions on the Troubleshooting page:
https://github.com/ramensoftware/windhawk/wiki/Troubleshooting#some-or-all-windhawk-mods-dont-work

> * Disable the mod.
> * Go to the mod's Advanced tab.
> * Set **Debug logging** to **Detailed debug logs**.
> * Click on **Show log output**.
> * Enable the mod.
> * Verify that the mod doesn't work, copy the log and save it to a file.
> * Set **Debug logging** back to **None**.

**Comment by LesFerch** [comment]
After KB5034204 is installed, disabling feature number 40729001 no longer has any effect. Therefore, the ViveTool method to enable the old Explorer interface no longer works. I've also confirmed that ExplorerPatcher no longer works. If WindHawk is using the same method, there's your root cause. Given that information, are you still interested in seeing a debug log?

**Comment by m417z** [comment]
It won't hurt, and also, which of the two explorer styles are affected? (Classic navigation bar, Classic ribbon UI (no tabs))

**Comment by LesFerch** [comment]
OK, but I expect the log will just show that the code ran fine without any errors.

I just tested again and found that the "Classic ribbon UI" option also no longer works. It did switch initially, but the next open and close of Explorer was back to the App SDK version. Subsequent tests didn't switch at all. Restarting the Explorer task made no difference. This was unexpected because the Windows 10 ribbon option continues to work in StartAllBack and ExplorerPatcher and is available without any patch program with a [simple HKCU registry change](https://www.elevenforum.com/t/restore-classic-file-explorer-with-ribbon-in-windows-11.620/#Three).

I see the same effect with the "Classic navigation bar" option. The first launch of Explorer after applying the mod, sometimes starts up in the desire mode, but closing and opening Explorer then results in an Explorer window with no address / search bar at all.

**Comment by LesFerch** [comment]
This is all that appears in the log when "Classic navigation bar" is selected and enabled:
```
DebugViewConsole 1.8.0.102
Listening for OutputDebugString messages...
```

Here's the log when "Classic ribbon UI" is selected and enabled:

[ClassicRibbonUI.log](https://github.com/ramensoftware/windhawk-mods/files/14067509/ClassicRibbonUI.log)

**Comment by m417z** [comment]
I installed the update and tried both options. "Classic ribbon UI" didn't work for me, but "Classic navigation bar" did. I closed and re-opened explorer multiple times and it seems fine.

> Here's the log when "Classic ribbon UI" is selected and enabled:
> 
> [ClassicRibbonUI.log](https://github.com/ramensoftware/windhawk-mods/files/14067509/ClassicRibbonUI.log)

It can be seen from the logs that the function that's hooked for this option, "WASDKInFileExplorer", no longer exists. Thanks for the registry pointer, perhaps I can use it for an alternative implementation.

> This is all that appears in the log when "Classic navigation bar" is selected and enabled:
> 
> ```
> DebugViewConsole 1.8.0.102
> Listening for OutputDebugString messages...
> ```

This empty log doesn't make sense, there must be something else when the mod is enabled. Can you please try it again?

**Comment by LesFerch** [comment]
I reset my VM to Windows 11 23H2 build 22631.2428 and retested both "Classic navigation bar"  and "Classic ribbon UI". Both options worked perfectly.

I then installed the KB5034204 update (build 22631.3085) and tested again. This time, my results matched yours. The "Classic ribbon UI" option did not work, but the "Classic navigation bar" worked fine.

I thought the problem may have been related to tests I did with ExplorerPatcher and ViveTool, but I ran those tests again and the WindHawk patch continued to work for the "Classic navigation bar". I have no idea now what the problem was in my VM before I reset it.

So, I'll change the text in the title to "Classic ribbon UI" and that may be an easy fix given that HKCU registry entry.
